### ISSUE TITLE
JSON Patch format YAML files - yaml validation error: yaml: unmarshal errors: cannot unmarshal !!seq into map[string]interface {}

### ISSUE THREAD

**Issue by HariSekhon** [datreeio/datree#633]
**Describe the bug**

JSON Patch yaml files cause the following error:

```
[X]  yaml validation error: yaml: unmarshal errors:
  line 29: cannot unmarshal !!seq into map[string]interface {}
```

**To Reproduce**
```
git clone https://github.com/HariSekhon/Kubernetes-configs k8s
cd k8s
datree test *.jsonpatch.yaml
```

Live example can be seen here in my GitHub Actions run:

https://github.com/HariSekhon/Kubernetes-configs/runs/6490327637?check_suite_focus=true#step:5:15

**Expected behavior**

Expected to handle json patches for k8s, or ignore them.

**Datree version (run `datree version`):**

Latest in GitHub Actions datreeio/action-datree

**Comment by adifayer** [comment]
@HariSekhon 
Regarding the unmarshal error, we will look into it and handle it. 
Moreover, I'm curious if you're using the kustomize patch mechanism. If this is the case- we officially support Kustomize, and it requires you to use a [different test command](https://hub.datree.io/integrations/kustomize-support).

**Comment by HariSekhon** [comment]
Yes these are kustomize patches using the standard JSON Patch construct.

However, since this repo is full of regular independent k8s manifests as well as some Kustomization files, it would be good if Datree could determine / ignore the different formats appropriately, at least for each command.

**Comment by eyarz** [comment]
if you want to skip "none-k8s" files, you can use the flag `--only-k8s-files`:
```
datree test *.yaml --only-k8s-files
```

**Comment by HariSekhon** [comment]
I already use `--only-k8s-files` in my CI workflow anyway, and was debugging the problem which occurred there, but just to confirm again:

```shell
$ datree test *.yaml --only-k8s-files
>>  File: /Users/hari/github/k8s/argocd-resources.argocd-application-controller.jsonpatch.yaml

[X] YAML validation

[X]  yaml validation error: yaml: unmarshal errors:
  line 24: cannot unmarshal !!seq into map[string]interface {}

...
```

**Comment by shmuelsa** [comment]
Hey @HariSekhon, 
fyi the issue was fixed in PR #644 and will be released shortly 

cheers!
